xapian-bindings/java/java.i

```diff
@@ -174,8 +174,11 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%ignore MSetIterator::next();
+%javamethodmodifiers MSetIterator::next(int) "private"
+
 %extend MSetIterator {
-    Xapian::docid next () {
+    Xapian::docid next (int pseudoArg) {
```

Comment:
The Java documentation seems to say that `next()` should throw an exception in this case:

https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html#next--

I suspect in practice people don't rely on that, but I don't know much about Java norms.  But running the C++ iterator past the end results in undefined behaviour, which is unhelpful in a case which Java defines the behaviour of.

At the C++ level, `iterator_valid()` is a very cheap check - it's just comparing an integer with zero:

```
inline bool iterator_valid(const Xapian::MSetIterator & it) {
    return it.off_from_end != 0;
}
```